src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,131 @@
-			while ((s >= 1) ) {
-				// Marsaglia polar for gaussian
-				x1 = uniform_real_dist(m_prng);
-				x2 = uniform_real_dist(m_prng);
-				s=x1*x1+x2*x2;
-			}
-
-			// =  x1/std::sqrt(val)* std::sqrt(-2*std::log(val));
-			randomcoeff_multiplicative[i * cur_dim_input_space + k] =
-			    x1 * std::sqrt(-2 * std::log(s) / s) / kernelwidth;
-		}
-	}
-
-	io::info("finished: initializing randomcoefficients ");
-
-	return true;
+void RandomFourierGaussPreproc::set_kernelwidth(float64_t kernelwidth) 
+{
+	require(m_kernelwidth > 0, "Kernelwidth must be greater than zero");
+	m_kernelwidth = kernelwidth;
 }
 
-void RandomFourierGaussPreproc::get_randomcoefficients(
-		float64_t ** randomcoeff_additive2,
-		float64_t ** randomcoeff_multiplicative2, int32_t *dim_feature_space2,
-		int32_t *dim_input_space2, float64_t* kernelwidth2) const {
-
-	ASSERT(randomcoeff_additive2)
-	ASSERT(randomcoeff_multiplicative2)
-
-	if (!test_rfinited()) {
-		*dim_feature_space2 = 0;
-		*dim_input_space2 = 0;
-		*kernelwidth2=1;
-		*randomcoeff_additive2 = NULL;
-		*randomcoeff_multiplicative2 = NULL;
-		return;
-	}
-
-	*dim_feature_space2 = cur_dim_feature_space;
-	*dim_input_space2 = cur_dim_input_space;
-	*kernelwidth2=cur_kernelwidth;
-
-	*randomcoeff_additive2 = SG_MALLOC(float64_t, cur_dim_feature_space);
-	*randomcoeff_multiplicative2 = SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive, randomcoeff_additive+cur_dim_feature_space,
-			*randomcoeff_additive2);
-	std::copy(randomcoeff_multiplicative, randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,
-			*randomcoeff_multiplicative2);
-
-
+float64_t RandomFourierGaussPreproc::get_kernelwidth() const noexcept 
+{
+	return m_kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_randomcoefficients(
-		float64_t *randomcoeff_additive2,
-		float64_t * randomcoeff_multiplicative2,
-		const int32_t dim_feature_space2, const int32_t dim_input_space2, const float64_t kernelwidth2) {
-	dim_feature_space = dim_feature_space2;
-	dim_input_space = dim_input_space2;
-	kernelwidth=kernelwidth2;
-
-	SG_FREE(randomcoeff_multiplicative);
-	randomcoeff_multiplicative=NULL;
-	SG_FREE(randomcoeff_additive);
-	randomcoeff_additive=NULL;
-
-	cur_dim_feature_space=dim_feature_space;
-	cur_dim_input_space = dim_input_space;
-	cur_kernelwidth=kernelwidth;
-
-	if( (dim_feature_space>0) && (dim_input_space>0) )
-	{
-	randomcoeff_additive=SG_MALLOC(float64_t, cur_dim_feature_space);
-	randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	std::copy(randomcoeff_additive2, randomcoeff_additive2
-			+ dim_feature_space, randomcoeff_additive);
-	std::copy(randomcoeff_multiplicative2, randomcoeff_multiplicative2
-			+ cur_dim_feature_space*cur_dim_input_space, randomcoeff_multiplicative);
-	}
+void RandomFourierGaussPreproc::init_basis(int32_t dim_input_space)
+{
+	io::info("generating random coefficients {}x{}",m_dims,dim_input_space);
```

Comment:
watch your whitespaces here. Also Maybe start info messages with a capital letter and end them with a period. Also why not "Generating {} random coefficients for input space dimension {}."